src/main/java/com/massivecraft/factions/util/FlightUtil.java

```diff
@@ -54,32 +54,58 @@ public void run() {
             for (Player player : Bukkit.getOnlinePlayers()) {
                 FPlayer pilot = FPlayers.getInstance().getByPlayer(player);
                 if (pilot.isFlying() && !pilot.isAdminBypassing()) {
-                    if (enemiesNearby(pilot, 5)) {
-                        pilot.msg(TL.COMMAND_FLY_ENEMY_DISABLE);
-                        pilot.setFlying(false);
-                        if (pilot.isAutoFlying()) {
-                            pilot.setAutoFlying(false);
+                    findAndDisable(pilot, 5);
+                }
+            }
+        }
+
+        private void findAndDisable(FPlayer target, int radius) {
+            final List<Entity> nearbyEntities = target.getPlayer().getNearbyEntities(radius, radius, radius);
+            for (Entity entity : nearbyEntities) {
+                if (entity instanceof Player) {
+                    final FPlayer playerNearby = FPlayers.getInstance().getByPlayer((Player) entity);
+                    if (playerNearby.isAdminBypassing()) {
+                        continue;
+                    }
+                    if (playerNearby.getRelationTo(target) == Relation.ENEMY) {
+                        if (target.isFlying()) {
+                            disableFlight(target);
+                        }
+
+                        if (playerNearby.isFlying()) {
+                            disableFlight(playerNearby);
                         }
                     }
                 }
             }
         }
 
         public boolean enemiesNearby(FPlayer target, int radius) {
```

Comment:
Maybe this and the new method which are almost the same could be merged, with just an added boolean to differentiate?